src/mscorlib/src/System/Lazy.cs

```diff
@@ -27,12 +27,81 @@
 
 namespace System
 {
-    // Lazy<T> is generic, but not all of its state needs to be generic.  Avoid creating duplicate
-    // objects per instantiation by putting them here.
-    internal static class LazyHelpers
+    class LazyImplementation
     {
-        // Dummy object used as the value of m_threadSafeObj if in PublicationOnly mode.
-        internal static readonly object PUBLICATION_ONLY_SENTINEL = new object();
+        public readonly static LazyImplementation NoneViaConstructor            = new LazyImplementation(LazyThreadSafetyMode.None, true);
+        public readonly static LazyImplementation NoneViaFactory                = new LazyImplementation(LazyThreadSafetyMode.None, false);
+        public readonly static LazyImplementation PublicationOnlyViaConstructor = new LazyImplementation(LazyThreadSafetyMode.PublicationOnly, true);
+        public readonly static LazyImplementation PublicationOnlyViaFactory     = new LazyImplementation(LazyThreadSafetyMode.PublicationOnly, false);
+        public readonly static LazyImplementation PublicationOnlySpinWait       = new LazyImplementation();
+
+        private readonly LazyThreadSafetyMode m_mode;
```

Comment:
We try to follow the [corefx coding style](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/coding-style.md) for new or significantly modified code. You can use just `_` as instance variable prefix.